pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -36,11 +36,15 @@ def pytorch_typify_tensor(data, dtype=None, **kwargs):
 
 @pytorch_typify.register(slice)
 @pytorch_typify.register(NoneType)
-@pytorch_typify.register(np.number)
 def pytorch_typify_no_conversion_needed(data, **kwargs):
     return data
 
 
+@pytorch_typify.register(np.number)
+def pytorch_typify_extract(data, **kwargs):
+    return data.item()
```

Comment:
This seems wrong?

Reply:
Say more? 

The torch compiler threw asserts when a zero dim np value was passed back.

Reply:
Isn't it going to upcast values to float64, whatever python integers are? Does torch have scalars (not tensors) with specific dtypes we can use instead?